File: cal.swift
import Foundation

func calculate(expression: String) -> Double? {
    let exp = NSExpression(format: expression)
    if let result = exp.expressionValue(with: nil, context: nil) as? Double {
        return result
    }
    return nil
}

print("Enter an expression: ", terminator: "")
if let expression = readLine() {
    if let result = calculate(expression: expression) {
        print("Result: \(result)")
    } else {
        print("Invalid expression")
    }
}

File: cal.php
<?php
function calculate($expression) {
    if (preg_match('/^[0-9+\-*/().\s]+$/', $expression)) {
        $result = @eval("return $expression;");
        if ($result === false) {
            throw new Exception("Invalid expression");
        } else {
            return $result;
        }
    } else {
        throw new Exception("Invalid characters in expression");
    }
}

if (php_sapi_name() == "cli") {
    echo "Enter an expression: ";
    $expression = trim(fgets(STDIN));
    $result = calculate($expression);
    echo "Result: $result\n";
} else {
    if (isset($_GET['expression'])) {
        $expression = $_GET['expression'];
        $result = calculate($expression);
        echo "Result: $result";
    } else {
        echo '<form method="GET">
                <input type="text" name="expression" placeholder="Enter an expression" />
                <button type="submit">Calculate</button>
              </form>';
    }
}
?>